File: backend/routes/raycheck.ts
import { Router } from "express";
import { getMapConfiguration, computeRays, getBuildingsData } from "../controllers/raycheck";
const router = Router();
router.get("/:mapTitle", getMapConfiguration);
router.get("/buildings/:mapTitle", getBuildingsData);
router.post("/compute", computeRays);

export default router;

File: backend/routes/raycheck.go
package routes

import (
	"backendGo/controllers"

	"github.com/gin-gonic/gin"
)

func SetupRayCheckRoutes(router *gin.Engine) {
	raycheckRouter := router.Group("/api/maps")
	{
		raycheckRouter.GET("/", controllers.GetMaps)
		raycheckRouter.GET("/:mapTitle", controllers.GetMapById)
		raycheckRouter.POST("/rayLaunch/:mapTitle", controllers.Create3DRayLaunching)
	}
}
